### ISSUE TITLE
Deep Linking Not Working as Expected with App Caching in Microsoft Teams Tab Application

### ISSUE THREAD

**Issue by TahaKeles** [OfficeDev/microsoft-teams-library-js#1933]
I have implemented app caching for my Microsoft Teams Tab application by following [this official documentation](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/app-caching). The caching mechanism works as expected on initial installation; however, I am experiencing issues when trying to use deep linking with adaptive cards.

Expected Behavior:

When I click an adaptive card that should deep-link into a specific section of my application, the deep linking should work as expected.

Current Behavior:

Clicking on an adaptive card leads me to the location where the application was last cached. The contentUrl and entityId are not updating to handle the deep-linking scenario. They remain as they were initially set for caching. I've checked the output of my code block, and it confirms that it's identical before and after attempting deep linking, which shouldn't be the case.

Steps to Reproduce:

Implement app caching as per [this guide](https://learn.microsoft.com/en-us/microsoftteams/platform/tabs/how-to/app-caching).
Navigate to different sections within the app to ensure caching is working.
Trigger an adaptive card that attempts to deep-link to a specific section in the application.
Observe that you are led to the last cached section rather than the intended deep-link target.
Check code output, and note that it remains the same after deep linking, confirming the issue.

Question:

How can I update the contentUrl or entityId or any other related parameters to make deep linking compatible with my cached application?

Additional Screenshots:

<img width="569" alt="Screenshot1" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/44381293/1ee025b9-9f0f-4204-bb07-16115f561ebf">

<img width="664" alt="Screenshot2" src="https://github.com/OfficeDev/microsoft-teams-library-js/assets/44381293/cbabc6a5-9ff5-4793-94a3-de265ca83229">

Environment:

Microsoft Teams
Language: TypeScript
Framework: Node.js, React

Any help to resolve this issue would be highly appreciated. Thank you!

**Comment by vikramtha** [comment]
Hi @TahaKeles,

Just to better understand your situation, you are saying that the deeplink leads you to the last opened tab/page when the application was open and not to the deeplink target which is a different tab/page when caching is enabled?

**Comment by TahaKeles** [comment]
Yes, you've understood the issue correctly. When caching is enabled and I attempt to use a deep link to navigate to a specific tab/page within the application, it leads me to the last opened tab/page that was cached, rather than the intended deep-link target.

Thank you for looking into this. Would appreciate any guidance on how to resolve this.

**Comment by nwojod-MSFT** [comment]
@TahaKeles thanks for the additional context. We will investigate this issue on our end and share update with you.

**Comment by sayali-MSFT** [comment]
@TahaKeles - Thanks for reporting your issue.
We will check this at our end and will get back to you.

**Comment by sayali-MSFT** [comment]
We are trying to reproduce the issue and encountering some difficulties while doing so. We will update you as soon as we are able to replicate it.

**Comment by sayali-MSFT** [comment]
@TahaKeles -Is the issue happening with new teams or classic teams?
